### ISSUE TITLE
Severity standardization - Conditional on user mistake

### ISSUE THREAD

**Issue by trust1995** [code-423n4/org#53]
It would be beneficial for all of us to standardize severities for common issues, as we have seen variance among judges and contests.

Example of two different severities:
[Frax Liquidity contest](https://github.com/code-423n4/2022-09-frax-findings/issues/18) - Using unsafe transfer to sweep mistakenly sent tokens, means unsupported tokens like USDT are stuck in contract. Judged M.
[Blur contest](https://github.com/code-423n4/2022-10-blur-findings/issues/723) - theoretic leak of value - if seller supplies ETH to the execute() operation, they will pay instead of buyer

We need to have a fixed policy on how requiring user mistake reduces severity level. 
One simple option is - if the API is used in a way that is not sanctioned by the protocol, the MAX severity would is Low. 
Another option is - if it is deemed likely by the judge that an API will be used in some particular, wrong way, and there is a lack of check for this wrong usage which may lead to user losses, it could be accepted as a Medium.

I lean towards the first option as it levels the playing field, makes it easier for judges, and cannot lead to very subjective disputes. The downside is that sponsors will get less reports for safety checks that they should add, it is unclear if these would find their way to QA reports.

Opinions are welcome.
@GalloDaSballo

**Comment by GalloDaSballo** [comment]
Specific counter-argument, that only shipped projects will experience is that many new projects will airdrop their coins to your Strategy / Pool and not having a sweep means hundreds of people getting pissed for losing an airdrop. (Some airdrops are worth thousands of dollars per person)

I think if the caller is sending funds because they make a mistake, Low is appropriate. Especially if the code makes it clear that's an unintended path which is possible.

However, I think there are cases where a Loss of Yield can happen (e.g. airdrops / claims) and so DeFi Pools should always consider having a sweep function.

I'm ambivalent, and I think I could have rated Med for Frax or Low, perhaps based on the Sponsor or other Warden's Feedback.


TL;DR I don't have a strong opinion and believe some Contracts not having Sweep should be escalated (e.g. Defi Strategies)
While I think random people sending coins should not warrant a Med Severity.

The reason why Blur finding should receive a low is that the code does check in the other case, meaning you can argue the developer **wanted** to have the check, but didn't / forgot, which is different from point blank saying that the contracts are responsible for people making mistakes

**Comment by trust1995** [comment]
This is a good discussion, we can further develop a specific list of mistake-related issues that could be accepted as medium.
ERC20 Sweeping functions, specifically for DeFi pools / marketplaces, are probably important enough to deserve M.

**Comment by nine-december** [comment]
Pretty interesting discussion. Regarding sanity checks with `address(0)` I believe that it is correct to consider them as QA. 

However, on functions that handle value in two different ways depending on how and what the users input ensuring that no value is mistakenly sent while using tokens as payment differs from the sanity check mentioned before. The volume of calls that the leaking function will receive as well as the value it could handle increase the likelihood of failure to a certain level were a single mistakenly made call can cause an irreversible leak of value. 

Marketplaces and Defi pools are meant to handle tokens and they must reduce the chances on which users can leak value, even if it was a user mistake. If we strive for massive adoption of this technology, we must protect users even if they do not know how to sign a transaction. 

As for the field I come, chemical process engineering, having a process that leaks value is a problem that could be also dangerous regardless if operators or end consumers do not act as they meant. It is our job as engineers to reduce the chances for our process to leak. 

That's my humble POV.

**Comment by dmvt** [comment]
> As for the field I come, chemical process engineering, having a process that leaks value is a problem that could be also dangerous regardless if operators or end consumers do not act as they meant. It is our job as engineers to reduce the chances for our process to leak.

I agree with this sentiment. It is not reasonable or possible to protect the user from every mistake they can make. However, when the value lost is guaranteed to be very large and preventing the mistake has a very low cost, it is reasonable to assign a medium risk rating. The closer the cost/benefit ratio gets to zero, the more likely the issue should be rated QA. Further, if the suggested mitigation would not prevent the error in most cases (like address(0) checks), the issue should likely be QA.

**Comment by CloudEllie** [comment]
Per the Autumn 2023 C4 Supreme Court verdicts, the Supreme Court's verdict on this issue is: 

> Findings that require the user to be careless or enter the wrong information into a contract call are QA or Invalid. 
> Non privileged users are expected to preview their transactions to protect against input mistakes. Anyone using modern tooling will have previews built into their toolset.
> Phishing attacks, and improper caution on using a protocol fall under this rule.

Link to verdict: https://docs.google.com/document/d/1Y2wJVt0d2URv8Pptmo7JqNd0DuPk_qF9EPJAj3iSQiE/edit#heading=h.lot96e9mx5tv
